docs/04-reference/winglang-spec.md

```diff
@@ -869,30 +869,223 @@ let f = (arg1: num, var arg2: num) => {
 
 ### 1.7 Optionality
 
-Symbol `?` can mark a type as optional.  
-Optionality means the value behind type can be either present or nil.
+Nullity (`null`, `undefined`, `nil`) is a primary source of bugs in software.
+Being able to guarantee that a value will never be null makes it easier to write safe
+code without constantly having to take nullity into account.
 
-Rules of optionality apply to the entire new container type of `type?` and not
-the value behind it (`type`).  
+We are going to try and design Wing without allowing users to explicitly assign or use the `nil`
+value. Instead, the language offers syntax to support the various use cases that arise when dealing
+with *optionality*.
 
-Using the `??` operator allows one to safely access the value behind an optional
-variable by always providing a default, in case the variable is nil. This forces
-a value to be present for the l-value (left hand side of the assignment operator
-and guarantees what's returned is type-stripped from its `?` keyword).
+Here's a quick summary of how optionality works in Wing:
 
-> ```TS
-> let x? = 44;
-> let y = x ?? 55;
-> ```
+* `x: T?` marks `x` as "optional of T". This means that `x` can store a value of type `T` but it can also be in a state where there is no value stored there.
+* To test if `x` has a value or not, use `x?`, which returns a `bool`. If x has a value, `x?` will be `true`, otherwise it will be `false`.
+* The `if let y = x { } else { }` is a special if statement that binds `y` inside the block only if `x` includes a value. Otherwise, the `else` block will be executed.
+* The `x?.y` notation can be used to access object fields only if there is a value. It returns an empty optional if the value is not defined.
+* The `x ?? y` notation will return the value in `x` (if there is one) or `y` otherwise (`y` must be of type `T`, not `T?`).
+* A default value can also be supplied using `= y` in the declaration of the field or argument, which implies that the type will always be defined.
```

Comment:
by
> using `= y`

Do you mean that these are valid:
```wing
let x: num? = 7;
let var y: num? = 7; // `y` is `var` so it might not always be defined.
```
Also if the optional variable is marked `var` then this might not be true:
> which implies that the type will always be defined

Reply:
No, in struct fields:

```js
struct Foo {
  age: num = 20;
}
```

And in function arguments:

```js
let print_age = (age: num = 88) => {
  print("my age: ${age}");
};
```

Added clarification.